code/GameEngine/GameObject/GameObject.cs

```diff
@@ -232,6 +232,20 @@ public IEnumerable<T> GetComponents<T>( bool enabledOnly = true, bool deep = fal
 		}
 	}
 
+	public bool TryGetComponent<T>( out T component )
+	{
+		component = GetComponent<T>();
+
+		return component is not null;
+	}
+
+	public bool TryGetComponents<T>( out IEnumerable<T> components )
```

Comment:
I'm not sure the plural makes sense here

Reply:
Not really sure what else I'd call it, since it's grabbing all the components of that type.

Reply:
No, I don't know why it would exist. You'd just iterate GetComponents<T>, why would you ever want to TryGet the IEnumerable

Reply:
Good point, just overthinking it I guess. I'll push a fix in a sec